### PR TITLE
improvement(ics_space_amplification_goal_test.py): Added test for space-amplification-goal parameter

### PR DESCRIPTION
**Pull request by yarongilor**
IMPORTANT NOTE: most of the helper functions code in this test is already reviewed and merged, taken from the previous test of ics_space_amplification_test.py. it is done in purpose of differentiating old and new testing requirements.

this PR aims to meet only the testing requirements of: 
https://docs.google.com/document/d/1Fc4j0ICtQpN7QyPKQP_AqPa_UoanHEAEHm5EjOqPJ2Q/edit?usp=sharing

## PR pre-checks (self review)

- [x] I followed [KISS principle](https://en.wikipedia.org/wiki/KISS_principle) and [best practices](https://docs.google.com/document/d/1jihgOKb5iGRlD8_HQ92O0JbLk1kASUoZT23i_MXFSKI)
- [x] I didn't leave commented-out/debugging code
- [x] I added the relevant `backport` labels
- [ ] New configuration option are added and documented (in `sdcm/sct_config.py`)
- [ ] I have added tests to cover my changes (Infrastructure only - under `unit-test/` folder)
- [ ] All new and existing unit tests passed (CI)
- [ ] I have updated the Readme/doc folder accordingly (if needed)

### CODE REVIEW

**Comment by yarongilor** [comment]
A running job is found at:
https://jenkins.scylladb.com/job/scylla-staging/job/longevity-tests-ICS/job/SAG_performance_test/

File: ics_space_amplification_goal_test.py [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+GB_SIZE = MB_SIZE * 1024
+FS_SIZE_METRIC = 'node_filesystem_size_bytes'
+FS_SIZE_METRIC_OLD = 'node_filesystem_size'
+AVAIL_SIZE_METRIC = 'node_filesystem_avail_bytes'
+AVAIL_SIZE_METRIC_OLD = 'node_filesystem_avail'
+
+
+class IcsSpaceAmplificationTest(LongevityTest):
+
+    def _get_used_capacity_gb(self, node):  # pylint: disable=too-many-locals
```
Comment (fruch) [inline ics_space_amplification_goal_test.py:20]: all those functions are not needed in other tests ? only in this one ?
why they aren't part of cluster of tester ?

File: ics_space_amplification_goal_test.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+            # (3) measure space amplification for the re-written data
+            self.measure_nodes_space_amplification_after_write(
+                dict_nodes_initial_capacity=dict_nodes_capacity_before_overwrite_data,
+                written_data_size_gb=total_data_to_overwrite_gb, start_time=start_time)
+
+        InfoEvent(message=f"Space-amplification-goal testing cycles are done.")
+
+
+def generate_node_capacity_query_postfix(node):
+    node_ip = node.private_ip_address  # due to pylint bug
```
Comment (fruch) [inline ics_space_amplification_goal_test.py:229]: which bug is that ?

File: ics_space_amplification_goal_test.py [hunk 3]
```diff
@@ -0,0 +1,232 @@
...
+    scylla_dir = SCYLLA_DIR  # due to pylint bug
+    return '{{mountpoint="{scylla_dir}", instance=~".*?{node_ip}.*?"}}'.format(
```
Comment (fruch) [inline ics_space_amplification_goal_test.py:231]: why aren't you using f-strings here ?

File: ics_space_amplification_test.py [hunk 4]
```diff
@@ -17,7 +17,9 @@
 class IcsSpaceAmplificationTest(LongevityTest):
 
     def _get_used_capacity_gb(self, node):  # pylint: disable=too-many-locals
-        # example: node_filesystem_size_bytes{mountpoint="/var/lib/scylla", instance=~".*?10.0.79.46.*?"}-node_filesystem_avail_bytes{mountpoint="/var/lib/scylla", instance=~".*?10.0.79.46.*?"}
+        #  example: node_filesystem_size_bytes{mountpoint="/var/lib/scylla",
+        #  instance=~".*?10.0.79.46.*?"}-node_filesystem_avail_bytes{mountpoint="/var/lib/scylla",
+        #  instance=~".*?10.0.79.46.*?"}
         node_capacity_query_postfix = generate_node_capacity_query_postfix(node)
         filesystem_capacity_query = '{FS_SIZE_METRIC}{node_capacity_query_postfix}'.format(
```
Comment (fruch) [inline ics_space_amplification_test.py:24]: please use f-string here

File: ics_space_amplification_test.py [hunk 5]
```diff
@@ -27,7 +29,10 @@ def _get_used_capacity_gb(self, node):  # pylint: disable=too-many-locals
...
         fs_size_res = self.prometheus_db.query(query=filesystem_capacity_query, start=int(time.time()) - 5,
                                                end=int(time.time()))
-        assert fs_size_res, "No results from Prometheus"
+        if not fs_size_res:
+            self.log.warning(f"No results from Prometheus query: {filesystem_capacity_query}")
+            return 0
+            # assert fs_size_res, "No results from Prometheus"
         if not fs_size_res[0]:  # if no returned values - try the old metric names.
             filesystem_capacity_query = '{FS_SIZE_METRIC_OLD}{node_capacity_query_postfix}'.format(
                 **dict(locals(), **globals()))
```
Comment (fruch) [inline ics_space_amplification_test.py:38]: please switch to f-strings

File: ics_space_amplification_goal_test.py [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+
+class IcsSpaceAmplificationTest(LongevityTest):
+
+    def _get_used_capacity_gb(self, node):  # pylint: disable=too-many-locals
+        #  example: node_filesystem_size_bytes{mountpoint="/var/lib/scylla",
+        #  instance=~".*?10.0.79.46.*?"}-node_filesystem_avail_bytes{mountpoint="/var/lib/scylla",
+        #  instance=~".*?10.0.79.46.*?"}
+        node_capacity_query_postfix = generate_node_capacity_query_postfix(node)
+        filesystem_capacity_query = '{FS_SIZE_METRIC}{node_capacity_query_postfix}'.format(
+            **dict(locals(), **globals()))
```
Comment (fruch) [inline ics_space_amplification_goal_test.py:26]: f-strings

File: ics_space_amplification_goal_test.py [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+        used_capacity_query = '{filesystem_capacity_query}-{AVAIL_SIZE_METRIC}{node_capacity_query_postfix}'.format(
+            **dict(locals(), **globals()))
```
Comment (fruch) [inline ics_space_amplification_goal_test.py:28]: f-strings

File: ics_space_amplification_test.py [hunk 8]
```diff
@@ -2,12 +2,12 @@
 import time
```
Comment (yarongilor) [inline ics_space_amplification_test.py:2]: Changes to this file may temporarily be ignored until we confirm if there's still a need to keep it for backward comparisons

File: ics_space_amplification_goal_test.py (hunk 2 above)
Comment (yarongilor) [reply ics_space_amplification_goal_test.py:228]: perhaps a legacy issue. in anyway it's an existing master code..

File: ics_space_amplification_goal_test.py (hunk 1 above)
Comment (yarongilor) [reply ics_space_amplification_goal_test.py:20]: no other test is interested with nodes used capacity apparently.

File: ics_space_amplification_goal_test.py (hunk 3 above)
Comment (yarongilor) [reply ics_space_amplification_goal_test.py:230]: old code from master as well..

File: ics_space_amplification_test.py (hunk 4 above)
Comment (yarongilor) [reply ics_space_amplification_test.py:24]: existing master code

File: ics_space_amplification_test.py (hunk 5 above)
Comment (yarongilor) [reply ics_space_amplification_test.py:38]: existing master code

File: ics_space_amplification_goal_test.py (hunk 6 above)
Comment (yarongilor) [reply ics_space_amplification_goal_test.py:26]: existing master code

File: ics_space_amplification_goal_test.py (hunk 7 above)
Comment (yarongilor) [reply ics_space_amplification_goal_test.py:28]: existing master code

File: ics_space_amplification_goal_test.py (hunk 1 above)
Comment (fruch) [reply ics_space_amplification_goal_test.py:21]: I don't understand, you are on one hand telling this code isn't being used by any other test.

But on the other hand telling all the issue I've mentioned below like the not f-strings are cause of it copied from old code.

This suggests there are two test files using those functions, and you just duplicate them from on to other.

Let's move those utility functions to a place they can be shared, if that's on the Cluster object or the Tester object.

Comment (fruch) [reply ics_space_amplification_goal_test.py:21]: And handle all the things I've noted, while we at it.

Copying code, and not addressing those comments, just cause it's originated from older code, doesn't seem reasonable to me.

File: ics_space_amplification_test.py (hunk 8 above)
Comment (fruch) [reply ics_space_amplification_test.py:2]: Nothing is permanent as a temporary thing.

If it stays, need to be review like any other change.

If it goes away, let's delete it now. 

If you think it's irrelevant to this PR, undo the changes to it.

File: ics_space_amplification_goal_test.py (hunk 1 above)
Comment (yarongilor) [reply ics_space_amplification_goal_test.py:21]: previous test files were deleted.

File: test-cases/features/ics_space_amplification_goal_test.yaml [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+
+stress_cmd: ["cassandra-stress write no-warmup cl=QUORUM n=1000000000 -schema 'replication(factor=3) compaction(strategy=IncrementalCompactionStrategy)' -port jmx=6868 -mode cql3 native -rate threads=20 -col 'size=FIXED(200) n=FIXED(5)' -pop seq=1..1000000000"]
+round_robin: true
+
+n_db_nodes: 3
+n_loaders: 4
+n_monitor_nodes: 1
+
+instance_type_db: 'i3.4xlarge'
+ami_id_loader: 'ami-03c856a491257cb7f'
```
Comment (fruch) [inline test-cases/features/ics_space_amplification_goal_test.yaml:17]: @dkropachev is checking this for us again, let wait for his answer

**Comment by fruch** [comment]
@yarongilor please check why the autopep8 isn't happy

**Review by fruch — APPROVED** [review]
LGTM,
let just wait for @dkropachev answer regarding the load AMI,
and please see if the autopep8 issues are cause of you code

File: test-cases/features/ics_space_amplification_goal_test.yaml (hunk 9 above)
Comment (fruch) [reply test-cases/features/ics_space_amplification_goal_test.yaml:17]: We had any answer from him ?

**Comment by fruch** [comment]
Check again why autopep8 isn't happy

File: test-cases/features/ics_space_amplification_goal_test.yaml (hunk 9 above)
Comment (yarongilor) [reply test-cases/features/ics_space_amplification_goal_test.yaml:17]: no answer. sent him test log earlier.

**Comment by yarongilor** [comment]
> Check again why autopep8 isn't happy

it has the following report:
```
22:56:42  [file:ics_space_amplification_goal_test.py]
22:56:42  --->  Applying global fix for E265
22:56:42  --->  0 issue(s) to fix {}
```

**Pull request merged**